libs/core/src/components/sage-textarea/sage-textarea.tsx

```diff
@@ -0,0 +1,164 @@
+import { Component, Element, Host, h, Prop, Event, EventEmitter } from '@stencil/core';
+import { ChangeEvent } from 'react';
+import { isRequired } from '../../utils/utils';
+
+
+@Component({
+  tag: 'sage-textarea',
+  styleUrl: 'sage-textarea.scss',
+  shadow: true,
+})
+export class SageTextarea {
+  @Element() el: HTMLDivElement;
+
+  /**
+   * A unique identifier for the textarea
+   */
+  @Prop() componentId?: string;
+
+  /**
+   * Indicates whether or not the textarea is disabled
+   * @defaultValue false
+   */
+  @Prop() disabled = false;
+
+  /**
+   * Specifies the error text and provides an error-themed treatment to the field
+   */
+  @Prop() errorMessage?: string;
+
+  /**
+   * Displays a hint or description of the textarea
+   */
+  @Prop() hintMessage?: string;
+
+  /**
+   * Indicates whether or not the textarea is invalid or throws an error
+   * @defaultValue false
+   */
+  @Prop({mutable: true}) invalid = false;
+
+  /**
+   * Text to be displayed as the textarea label
+   */
+  @Prop() label?: string;
+
+  /**
+   * Specifies the name, submitted with the form name/value pair. This value will mirror the componentId
+   */
+  @Prop() name: string = this.componentId;
+
+  /**
+   * Specifies a short hint that describes the expected value of the textarea
+   */
+  @Prop() placeholder?: string;
+
+  /**
+   * Indicates whether or not the textarea is readonly
+   * @defaultValue false
+   */
+  @Prop() readonly = false;
+
+  /**
+   * Indicates whether or not the textarea is required
+   * @defaultValue false
+   */
+  @Prop() required = false;
+
+  /**
+   * Sets number of rows of text visible without needing to scroll in the textarea
+   */
+  @Prop() rows?: number;
+
+  /**
+   * The value of the textarea
+   */
+  @Prop({mutable: true}) value?: string;
+
+  /**
+   * Event emitted whenever the value of the textarea changes
+   */
+  @Event() sageTextareaChange: EventEmitter<ChangeEvent>;
+  private onTextareaInputEvent = (ev: Event) => {
+    const textarea = ev.target as HTMLTextAreaElement;
+    isRequired(textarea, this);
+    if (textarea) {
+      this.value = textarea.innerHTML;
+    }
+    this.sageTextareaChange.emit();
+  };
+
+  private textareaClassNames = () => {
+    let className = `sage-textarea__field`;
+    if (this.invalid && this.invalid === true) {
+      const invalidClassName = `is-invalid`;
+      className += ' ' + invalidClassName;
+    }
+    return className;
```

Comment:
Another approach could be used here that would remove the need for the `+=` and `' '`

```
private textareaClassNames() {
  const classNames = ['sage-textarea__field'];

  if (this.invalid && this.invalid === true) {
    classNames.push('is-invalid');
  }

  return classNames.join('  ');
}
```